Commonality Analysis Code.Rmd

```diff
@@ -0,0 +1,258 @@
+miss_var_summary(missing_pfs)
+#Looks like 1 participant (row 782) did not answer item 2, so need to drop them
+cope_ca_data_imputed <- cope_ca_data_imputed[-782, ]
+#Now at #1382 observations
+```
+
+```{r create new data frame of complete cases (nonimputed) - for robustness check later}
+selected_columns <- c("b_bhs", "b_shs", "b_cdi", "b_gad", "b_cts_rs", "b_drs_month", "pi_pfs", "pi_bhs", "pi_shs", "f1_bhs", "f1_shs", "f1_cdi", "f1_gad", "f1_cts_rs", "f1_drs_month")
+
+cope_ca_data_nonimputed <- cope_ca_data %>%
+  filter(complete.cases(across(contains(selected_columns))))
+#880 observations
+```
+
+```{r PFS correlation matrix}
+#For the imputed dataset
+pfs_imputed <- cope_ca_data_imputed %>%
+  select(pi_pfs_1, pi_pfs_2, pi_pfs_3, pi_pfs_4, pi_pfs_5, pi_pfs_6, pi_pfs_7)
+pfs_imputed_cor <- cor(pfs_imputed)
+pfs_imputed_cor
+pfs_imputed_cor_plot <- corrplot(pfs_imputed_cor)
+
+#For the nonimputed dataset
+pfs_nonimputed <- cope_ca_data_nonimputed %>% 
+  select(pi_pfs_1, pi_pfs_2, pi_pfs_3, pi_pfs_4, pi_pfs_5, pi_pfs_6, pi_pfs_7)
+pfs_nonimputed_cor <- cor(pfs_nonimputed)
+pfs_nonimputed_cor
+pfs_nonimputed_cor_plot <- corrplot(pfs_nonimputed_cor)
+
+#We pre-registered we would use the mean PFS scores in the regression models if correlations between items are >0.7. They are surprisingly <0.7, so it looks like we'll be doing a principal component analysis.
+```
+
+```{r principal component analysis}
+
+#Imputed data
+
+#We probably only want one component, so starting with nfactors = 1
+pca1 <- psych::principal(pfs_imputed, nfactors = 1, rotate = "none", scores = TRUE)
+pca1
+#SS loadings    3.29
+#Proportion Var 0.47
+cope_ca_data_imputed <- cbind(cope_ca_data_imputed, pca1$scores)
+#For some reason, getting multiple PC1 columns but with the same values
+
+#Seeing if more components results in way more variance being explained, so playing with nfactors = 7 (number of PFS items)
+pca2 <- psych::principal(pfs_imputed, nfactors = 7, rotate = "none")
+pca2
+plot(pca2$values, type="b")
+#Inflection point of the scree plot is the second component, so probably does not make sense to have more than one component
+
+cope_ca_data_imputed <- cbind(cope_ca_data_imputed, pca1$scores)
+
+#Nonimputed data
+#We probably only want one component, so starting with nfactors = 1
+pca3 <- psych::principal(pfs_nonimputed, nfactors = 1, rotate = "none")
+pca3
+#SS loadings    3.28
+#Proportion Var 0.47
+cope_ca_data_nonimputed <- cbind(cope_ca_data_nonimputed, pca3$scores) 
+#For some reason, getting multiple PC1 columns but with the same values
+
+#Seeing if more components results in way more variance being explained, so playing with nfactors = 7 (number of PFS items)
+pca4 <- psych::principal(pfs_nonimputed, nfactors = 7, rotate = "none")
+pca4
+plot(pca4$values, type="b")
+#Inflection point of the scree plot is the second component, so probably does not make sense to have more than one component
+```
+
+```{r stuff for Isaac}

```

Comment:
These all look good to me! You can probably delete this and summarize everything by just showing the correlation between the PFS mean score and the PC1

Reply:
We might just keep the scree plots for our own reference but we added the correlations and they are 0.99 for both the imputed and nonimputed datasets!